Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp

```diff
@@ -92,9 +92,61 @@ namespace AzGameFramework
     {
         m_headless = headless;
     }
+    void GameApplication::SetConsoleModeSupported(bool supported)
+    {
+        m_consoleModeSupported = supported;
+    }
 
     void GameApplication::StartCommon(AZ::Entity* systemEntity)
     {
+        if (!this->m_headless)
+        {
+            bool isConsoleMode{ false };
+
+            // If we are not in headless-mode, its still possible to be in console-only mode,
+            // where a native client window will not be created (on platforms that support 
+            // setting console mode at runtime). There are 2 possible triggers for console mode
+
+            // 1. Settings registry : If the registry setting '/O3DE/Launcher/Bootstrap/ConsoleMode' is specified and set to true
+            if (const auto* settingsRegistry = AZ::SettingsRegistry::Get())
+            {
+                settingsRegistry->Get(isConsoleMode, "/O3DE/Launcher/Bootstrap/ConsoleMode");
+            }
+
+            // 2. Either '-console-mode' or 'rhi=null' is specified in the command-line argument
+            const AzFramework::CommandLine* commandLine{ nullptr };
+            constexpr const char* commandSwitchConsoleOnly = "console-mode";
+            constexpr const char* commandSwitchRhi = "rhi";
+            AzFramework::ApplicationRequests::Bus::BroadcastResult(commandLine, &AzFramework::ApplicationRequests::GetApplicationCommandLine);
+            AZ_Assert(commandLine, "Unable to query application command line to evaluate console-mode switches.");
+            if (commandLine)
+            {
+                if (commandLine->HasSwitch(commandSwitchConsoleOnly))
+                {
+                    isConsoleMode = true;
+                } 
+                else if (size_t switchCount = commandLine->GetNumSwitchValues(commandSwitchRhi); switchCount > 0)
+                {
+                    auto rhiValue = commandLine->GetSwitchValue(commandSwitchRhi, switchCount - 1);
```

Comment:
I guess we would then need to also do this for -NullRenderer (which is used to disable RPI). Check RPISystemComponent::Activate for more information on how it is checked